source/Menu.h

```diff
@@ -41,29 +42,143 @@
+                     enet_uint32 publicAddress, enet_uint16 publicPort,
+                     enet_uint32 localAddress, enet_uint16 localPort,
+                     const std::string &address, const std::string &port,
+                     const std::string &publicAddressStr, const std::string &publicPortStr,
+                     const std::string &localAddressStr, const std::string &localPortStr,
+                     const std::string &descr, const std::string &text,
+                     bool enableNAT);
+            void clear();
+            void notify();
+            const std::vector<Server>& get() const;
+
+        };
+        class NetConfig {
+        private:
+            union hostAddress {
+                address_t address;
+                uint8_t a[4];
+            };
+        public:
+            // address of the master server
+            std::string masterServer = "duel6-master.mrakonos.cz";
+            Int32 masterServerPort = 5902;
+
+            // single switch to disable any communication with the master
+            // TODO (not implemented yet)
+            bool enableMasterServer = false;
+
+            // publish the game to master server for others to discover
+            bool enableMasterDiscovery = false;
+
+            // enable NAT traversal (server/client)
+            //  - client: false means we won't be able to connect to NATed servers
+            //  - server: false means we need public IP address for others to be able to contact us
+            bool enableNATPunch = false;
+
+
+            // local IP address for contacting the master server (in case we have more network interfaces in the computer)
+            std::string localIPAddress = "0.0.0.0";
+
+            std::string serverDescription = "unnamed server";
+
+            // experimental feature
+            // - for public server on a public IP address
+            // - but requiring a NAT handshake
+            // this is mainly for development purposes
+            std::string publicIPAddress = "0.0.0.0";
+            Int32 publicPort = 5900;
+            enet_uint32 getPublicIPAddress(){
+                hostAddress addr;
+                addr.address = 0;
+                if(publicIPAddress == "0.0.0.0" || publicIPAddress == "any" || publicIPAddress.length() > 15 || publicIPAddress.length() < 7){
+                    return 0;
+                }
+                auto a = 0;
+                std::string buffer;
+                buffer.reserve(16);
+                for(size_t i = 0 ; i < publicIPAddress.length(); i++){
+                    if(publicIPAddress.compare(i,1,".") == 0 && a < 4){
+                        addr.a[a++] = std::stoi(buffer);
+                        buffer.clear();
+                    } else {
+                        buffer += publicIPAddress[i];
+                    }
+                }
+                addr.a[a] = std::stoi(buffer);
+                return addr.address;
+            }
+        };
```

Comment:
Nebylo by hezci dat ty tridy do zvlastnich hlavickovych souboru, nez to mit vsechno tady